test/test-image.rb

```diff
@@ -323,5 +323,36 @@ def test_abs_diff
       assert_equal(expected.bytes.to_s,
                    @image.abs_diff(lined_image).bytes.to_s)
     end
+
+    def test_split
+      spliited_data = @image.bytes.to_s.each_char.group_by.with_index do |_, i|
+        i % 4
+      end
+      splitted_image_data = @image.split.collect do |splitted_image|
+        splitted_image.bytes.to_s
+      end
+      assert_equal([
+                     spliited_data[0].join(""),
+                     spliited_data[1].join(""),
+                     spliited_data[2].join(""),
+                     spliited_data[3].join(""),
+                   ],
+                   splitted_image_data)
+    end
+
+    def test_median_blur
+      blur_image = @image.median_blur(7)
+      assert_not_equal(@image.bytes.to_s,
+                       blur_image.bytes.to_s)
+    end
+
+    def test_median_blur_invalid_argument
+      # ksize must be odd and greater than 1
+      [-1,0,2].each do |n|
```

Comment:
Could you create a test for each case instead of looping in a test?
It's difficult to debug on failure because we can't run only the failed case.

We can use data driven test feature here (error message improvement is also included):

```diff
diff --git a/opencv-glib/image.cpp b/opencv-glib/image.cpp
index 00ad700..68ed77f 100644
--- a/opencv-glib/image.cpp
+++ b/opencv-glib/image.cpp
@@ -712,7 +712,8 @@ GCVImage *gcv_image_median_blur(GCVImage *image,
     g_set_error(error,
                 GCV_IMAGE_ERROR,
                 GCV_IMAGE_ERROR_FILTER,
-                "ksize must be odd and greater than 1");
+                "ksize must be odd and greater than 1: <%d>",
+                ksize);
     return NULL;
   }
 
diff --git a/test/test-image.rb b/test/test-image.rb
index 537a36d..ea03063 100644
--- a/test/test-image.rb
+++ b/test/test-image.rb
@@ -346,12 +346,12 @@ class TestImage < Test::Unit::TestCase
                        blur_image.bytes.to_s)
     end
 
-    def test_median_blur_invalid_argument
-      # ksize must be odd and greater than 1
-      [-1,0,2].each do |n|
-        assert_raise(CV::ImageError::Filter) do
-          @image.median_blur(n)
-        end
+    data(:ksize, [-1, 0, 2])
+    def test_median_blur_invalid_ksize(data)
+      ksize = data[:ksize]
+      message = "ksize must be odd and greater than 1: <#{ksize}>"
+      assert_raise(CV::ImageError::Filter.new(message)) do
+        @image.median_blur(ksize)
       end
     end
   end
```

See also: https://www.clear-code.com/blog/2018/12/26.html

Reply:
Thanks! Fixed.